lms/djangoapps/discussion/rest_api/serializers.py

```diff
@@ -475,48 +478,65 @@ class DiscussionSettingsSerializer(serializers.Serializer):
-            divided_course_wide_discussions = attrs.get(
+        course = self.context['course']
+        instance = self.context['settings']
+        if any(item in data for item in ('divided_course_wide_discussions', 'divided_inline_discussions')):
+            divided_course_wide_discussions, divided_inline_discussions = get_divided_discussions(
+                course, instance
+            )
+            divided_course_wide_discussions = data.get(
                 'divided_course_wide_discussions',
                 divided_course_wide_discussions
             )
-            divided_inline_discussions = attrs.get('divided_inline_discussions', divided_inline_discussions)
-            settings_to_change['divided_discussions'] = divided_course_wide_discussions + divided_inline_discussions
-
+            divided_inline_discussions = data.get('divided_inline_discussions', divided_inline_discussions)
+            payload['divided_discussions'] = divided_course_wide_discussions + divided_inline_discussions
         for item in ('always_divide_inline_discussions', 'division_scheme'):
-            if item in attrs:
-                settings_to_change[item] = attrs[item]
-        attrs['settings_to_change'] = settings_to_change
-        return attrs
+            if item in data:
+                payload[item] = data[item]
+        return payload
+
+    def to_representation(self, instance: CourseDiscussionSettings) -> dict:
+        """
+        Return a serialized representation of the course discussion settings.
+        """
+        payload = super().to_representation(instance)
+        course = self.context['course']
+        instance = self.context['settings']
+        course_key = course.id
+        divided_course_wide_discussions, divided_inline_discussions = get_divided_discussions(
+            course, instance
+        )
+        payload = {
+            'id': instance.id,
+            'divided_inline_discussions': divided_inline_discussions,
+            'divided_course_wide_discussions': divided_course_wide_discussions,
+            'always_divide_inline_discussions': instance.always_divide_inline_discussions,
+            'division_scheme': instance.division_scheme,
+            'available_division_schemes': available_division_schemes(course_key)
+        }
+        return payload
 
     def create(self, validated_data):
         """
-        Overriden create abstract method
+        This method intentionally left empty
         """
-        pass  # lint-amnesty, pylint: disable=unnecessary-pass
 
-    def update(self, instance, validated_data):
+    def update(self, instance: CourseDiscussionSettings, validated_data: dict) -> CourseDiscussionSettings:
         """
-        Overriden update abstract method
+        Update and save an existing instance
         """
-        pass  # lint-amnesty, pylint: disable=unnecessary-pass
+        try:
+            set_course_discussion_settings(course_key, **validated_data)
```

Comment:
Now we actually do the update inside `update`, instead of calling this helper directly.